pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/GenericSigBase.java

```diff
@@ -116,8 +121,9 @@ static class LazyClassSignature extends GenericSigBase<ClassStub> {
                            @Nullable String signature, // null if doesn't use generics in header
                            @Nullable String superInternalName, // null if this is the Object class
                            String[] interfaces) {
-            super(ctx);
+            super(ctx, "LazyClassSignature:" + ctx.getInternalName() + "[" + signature + "]");
             this.signature = signature;
+            this.typeParameterCount = GenericTypeParameterCounter.determineTypeParameterCount(this.signature);
```

Comment:
Here we parse partially the signature already to determine the type parameter count. That way, we know the number of types, without the need to know the exact types of the type parameters.